cluster-provision/gocli/providers/base_provider_test.go

```diff
@@ -35,6 +37,18 @@ var _ = Describe("Node Provisioning", func() {
 		}
 
 		k8sClient = k8s.NewTestClient(reactors...)
+		kp = NewKubevirtProvider("k8s-1.30", "", &client.Client{}, []KubevirtProviderOption{
```

Comment:
Could we get k8s-1.30 dynamically from what folders are existing? This will fail when removing the k8s-1.30 provider?

Reply:
No its unrelated to the folder and off the top of my mind it won't fail if we drop 1.30 altogether
the version validation logic has been moved to the run command, and since this is being defined in line then it wont be obstructed by this logic

`cluster-provision/gocli/cmd/run.go:143`
```golang
	allowedK8sVersions := []string{"k8s-1.28", "k8s-1.29", "k8s-1.30", "k8s-1.31"}
	var validVersion bool
	for _, v := range allowedK8sVersions {
		if k8sVersion == v {
			validVersion = true
		}
	}
	if !validVersion {
		return fmt.Errorf("Invalid k8s version passed, please use one of k8s-1.28, k8s-1.29, k8s-1.30 or k8s-1.31")
	}
```